From bugs+caf_=bugs=bugs.plt-scheme.org@plt-scheme.org Tue Mar 26 00:47:33 2013
Received: from mail-oa0-f49.google.com (mail-oa0-f49.google.com [209.85.219.49])
	by champlain.ccs.neu.edu (8.14.4/8.14.4) with ESMTP id r2Q4lUu4003161
	for <bugs@bugs.plt-scheme.org>; Tue, 26 Mar 2013 00:47:30 -0400
Message-Id: <201303260447.r2Q4lSdN003156@champlain.ccs.neu.edu>
Date: Tue, 26 Mar 2013 00:47:28 -0400
From: dyoo@hashcollision.org
To: bugs@racket-lang.org
Subject: exception showing when starting DrRacket

>Number:         13634
>Category:       all
>Synopsis:       exception showing when starting DrRacket
>Class:          sw-bug
>Responsible:    robby
>Severity:       serious
>Priority:       medium
>State:          closed
>Confidential:   no
>Arrival-Date:   Tue Mar 26 00:48:01 -0400 2013
>Closed-Date:    Tue Mar 26 01:04:02 -0400 2013
>Last-Modified:  Tue Mar 26 01:04:02 -0400 2013
>Originator:     Danny Yoo
>Organization:
plt
>Submitter-Id:   unknown
>Release:        5.3.3.7
>Environment:
MacIntel / Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_2) AppleWebKit/537.17 (KHTML, like Gecko) Chrome/24.0.1312.56 Safari/537.17
>Description:
I'm seeing the following exception when starting up DrRacket from 55ef0e7c0727ac417187f6e982cdd4c3791f760b:

128-110-64-140:build dyoo$ ~/local/racket/bin/drracket
DrRacket:module-language.rkt:first-opened exn: current-directory: `exists' access denied for /Users/dyoo/local/racket/src/build/
  (find-main-collects . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/setup/private/main-collects.rkt> 19 4 817 884))
  (#f . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/racket/private/promise.rkt> 104 10 5205 84))
  (call-with-exception-handler . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/racket/private/more-scheme.rkt> 263 2 9099 256))
  (force/generic . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/racket/private/promise.rkt> 96 0 4928 376))
  (exploded-root . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/setup/path-relativize.rkt> 31 4 1238 58))
  (#f . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/racket/private/promise.rkt> 104 10 5205 84))
  (call-with-exception-handler . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/racket/private/more-scheme.rkt> 263 2 9099 256))
  (force/generic . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/racket/private/promise.rkt> 96 0 4928 376))
  (path->relative . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/setup/path-relativize.rkt> 34 2 1371 938))
  (path->relative-string9 . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/setup/path-to-relative.rkt> 31 2 1132 805))
  (|[running body]| . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/unstable/wrapc.rkt> #f #f #f #f))
  (#f . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/racket/sandbox.rkt> 631 18 26381 211))
  (call-with-limits . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/racket/sandbox.rkt> 363 0 14665 1724))
  (evaluate-program . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/racket/sandbox.rkt> 611 0 25472 1345))
  (call-with-break-parameterization . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/racket/private/more-scheme.rkt> 146 2 4832 517))
  (user-process . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/racket/sandbox.rkt> 744 2 31367 2621))
  (#f . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/mred/private/wx/common/queue.rkt> 451 6 19918 1056))
  (#f . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/mred/private/wx/common/queue.rkt> 502 32 21905 120))
  (call-with-break-parameterization . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/racket/private/more-scheme.rkt> 146 2 4832 517))
  (loop . #(struct:srcloc #<path:/Users/dyoo/local/racket/collects/mred/private/wx/common/queue.rkt> 397 18 17838 540))
>How-To-Repeat:

>Fix:
>Audit-Trail:
Responsible changed from "nobody" to "robby" by robby@racket-lang.org at Tue, 26 Mar 2013 01:04:02 -0400
Reason>>> A commit by robby@racket-lang.org has resolved this report
  http://git.racket-lang.org/plt/commit/0c5fa7d6c5
State changed from "open" to "closed" by robby@racket-lang.org at Tue, 26 Mar 2013 01:04:02 -0400
Reason>>> A commit by robby@racket-lang.org has resolved this report
  http://git.racket-lang.org/plt/commit/0c5fa7d6c5
  | relax sandbox permissions now that contract error message
  | construction requires looking at the filesystem
  | 
  | assuming this is the right fix, it closes PR 13634

